File: program/find-the-eigenvectors-of-a-matrix/FindTheEigenvectorsOfAMatrix.java
import org.apache.commons.math3.linear.*;

public class Eigenvectors {
    public static void main(String[] args) {
        // Input matrix
        RealMatrix matrix = new Array2DRowRealMatrix(new double[][]{
            {1, 2, 3},
            {4, 5, 6},
            {7, 8, 9}
        });

        // Eigen decomposition
        EigenDecomposition decomposition = new EigenDecomposition(matrix);

        // Eigenvalues
        RealVector eigenvalues = decomposition.getRealEigenvalues();
        System.out.println("Eigenvalues: " + eigenvalues);

        // Eigenvectors
        RealMatrix eigenvectors = decomposition.getV();
        System.out.println("Eigenvectors:");
        for (int i = 0; i < eigenvectors.getColumnDimension(); i++) {
            RealVector eigenvector = eigenvectors.getColumnVector(i);
            System.out.println(eigenvector);
        }
    }
}

File: program/program/find-the-eigenvectors-of-a-matrix/FindTheEigenvectorsOfAMatrix.kt
import org.apache.commons.math3.linear.*

fun main(args: Array<String>) {
    val matrix: RealMatrix = Array2DRowRealMatrix(
        arrayOf(
            doubleArrayOf(1.0, 2.0, 3.0),
            doubleArrayOf(4.0, 5.0, 6.0),
            doubleArrayOf(7.0, 8.0, 9.0)
        )
    )

    val decomposition = EigenDecomposition(matrix)

    val eigenvalues: RealVector = decomposition.getRealEigenvalues()
    val eigenvectors: RealMatrix = decomposition.getV()

    for (i in 0 until eigenvectors.getColumnDimension()) {
        val eigenvector: RealVector = eigenvectors.getColumnVector(i)
        System.out.println(eigenvector)
    }
}
